File: leetcode.com/0945_minimum_increment_to_make_array_unique/solution.py
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        total = 0
        nums.sort()

        for i in range(len(nums) - 1):
            if nums[i] < nums[i + 1]:
                continue

            d = nums[i] - nums[i + 1] + 1

            nums[i + 1] += d
            total += d

        return total

File: leetcode.com/0945_minimum_increment_to_make_array_unique/solution.go
func minIncrementForUnique(nums []int) int {
    sort.Ints(nums)

    total := 0

    for i := 0 ; i < len(nums) - 1 ; i++ {
        if (nums[i] < nums[i + 1]) {
            continue
        }

        d := nums[i] - nums[i + 1] + 1
        total += d
        nums[i + 1] += d
    }

    return total
}
